mlflow/utils/_spark_utils.py

```diff
@@ -25,6 +36,9 @@ def _create_local_spark_session_for_recipes():
     except ImportError:
         # Return None if user doesn't have PySpark installed
         return None
+    # Callinig `_prepare_environ_for_creating_local_spark_session` is important,
+    # otherwise it might connect to parent process connected spark cluster.
```

Comment:
What does `it` refer to?

Reply:
it refers to current process.
The case is, suppose we have a parent process already initiate a spark session that connected to a spark cluster, then the parent process spawns a child process, if child process directly creates a local spark session, it does not work correctly, because of `PYSPARK_GATEWAY_PORT` and `PYSPARK_GATEWAY_SECRET` is inheriting from parent process and child process pyspark session will try to connect to the port and cause error.